justfile

```diff
@@ -246,9 +246,21 @@ wait-for-electrs-to-be-ready:
 
 build-ipa:
     #!/usr/bin/env bash
-    cd mobile && flutter build ipa --dart-define="ESPLORA_ENDPOINT=${ESPLORA_ENDPOINT}" \
-    --dart-define="COORDINATOR_P2P_ENDPOINT=${COORDINATOR_P2P_ENDPOINT}" --dart-define="COMMIT=$(git rev-parse HEAD)" \
-    --dart-define="BRANCH=$(git rev-parse --abbrev-ref HEAD)"
+    BUILD_NUMBER=$(git rev-list HEAD --count)
+    args=()
+
+    if [ "$NETWORK" = "regtest" ]; then
+      args+=(--flavor test)
+    fi
+
+    cd mobile && flutter build ipa "${args[@]}" \
+           --dart-define="ESPLORA_ENDPOINT=${ESPLORA_ENDPOINT}" \
+           --dart-define="COORDINATOR_P2P_ENDPOINT=${COORDINATOR_P2P_ENDPOINT}" \
+           --dart-define="NETWORK=${NETWORK}" \
+           --dart-define="COMMIT=$(git rev-parse HEAD)" \
+           --dart-define="BRANCH=$(git rev-parse --abbrev-ref HEAD)" \
+           --dart-define="COORDINATOR_PORT_HTTP=${COORDINATOR_PORT_HTTP}" \
+           --build-number=${BUILD_NUMBER}
```

Comment:
I wonder whether it wouldn't be simpler to ditch `.env.sample` and just have a separate `build-ipa` and build-ipa-regtest` targets, but that might be an overkill if it works for you as it is 🙃

Reply:
well.. I don't have a hard opinion of doing it either way. If you like to change that, go for it 🙂

I was thinking some implicit logic like this would be helpful on the pipeline. So that you don't have to provide an additional argument when running the build and release workflow manually.